### PR TITLE
Improve the logging messages to show container_id

### PR DESCRIPTION
**Pull request by rhatdan**
Also drop messages about skipping down to debug level, so admin can ignore them.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@jwhonce PTAL

**Comment by rhatdan** [comment]
Here is what the logs look like with this patch, on a leaked object
```
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <debug>: prestart container_id:ed630d533d0e30409ba4552b54716e2874c56205fa1692e818b73bdd13fc5060 rootfs:/var/lib/containers/storage/overlay/33d2c5abb2165ea2ad822c5045f0ae93beec713c2989e2b5f06323be2b3402d6/merged
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <debug>: ed630d533d0e30409ba4552b54716e2874c56205fa1692e818b73bdd13fc5060: Failed to canonicalize path [/var/lib/docker/overlay2]: No such file or directory. Skipping.
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <debug>: ed630d533d0e30409ba4552b54716e2874c56205fa1692e818b73bdd13fc5060: Failed to canonicalize path [/var/lib/docker/overlay]: No such file or directory. Skipping.
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <debug>: ed630d533d0e30409ba4552b54716e2874c56205fa1692e818b73bdd13fc5060: Failed to canonicalize path [/var/lib/docker-latest/overlay2]: No such file or directory. Skipping.
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <debug>: ed630d533d0e30409ba4552b54716e2874c56205fa1692e818b73bdd13fc5060: Failed to canonicalize path [/var/lib/docker-latest/overlay]: No such file or directory. Skipping.
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <debug>: ed630d533d0e30409ba4552b54716e2874c56205fa1692e818b73bdd13fc5060: Failed to canonicalize path [/var/lib/docker-latest/devicemapper]: No such file or directory. Skipping.
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <debug>: ed630d533d0e30409ba4552b54716e2874c56205fa1692e818b73bdd13fc5060: Failed to canonicalize path [/var/lib/docker-latest/containers/]: No such file or directory. Skipping.
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <debug>: ed630d533d0e30409ba4552b54716e2874c56205fa1692e818b73bdd13fc5060: Failed to canonicalize path [/var/lib/containers/storage/lvm]: No such file or directory. Skipping.
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <debug>: ed630d533d0e30409ba4552b54716e2874c56205fa1692e818b73bdd13fc5060: Failed to canonicalize path [/var/lib/containers/storage/devicemapper]: No such file or directory. Skipping.
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <info>: mapped host_mnt=/var/lib/docker/devicemapper to cont_mnt=/host/var/lib/docker/devicemapper
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <info>: ed630d533d0e30409ba4552b54716e2874c56205fa1692e818b73bdd13fc5060: Unmounted: [/var/lib/containers/storage/overlay/33d2c5abb2165ea2ad822c5045f0ae93beec713c2989e2b5f06323be2b3402d6/merged/host/var/lib/docker/devicemapper]
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <error>: Failed to unmount: [/var/lib/containers/storage/overlay/33d2c5abb2165ea2ad822c5045f0ae93beec713c2989e2b5f06323be2b3402d6/merged/host/var/lib/docker/devicemapper]
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <info>: mapped host_mnt=/var/lib/docker/containers to cont_mnt=/host/var/lib/docker/containers
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <info>: mapped host_mnt=/var/lib/containers/storage/overlay to cont_mnt=/host/var/lib/containers/storage/overlay
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <info>: ed630d533d0e30409ba4552b54716e2874c56205fa1692e818b73bdd13fc5060: Unmounted: [/var/lib/containers/storage/overlay/33d2c5abb2165ea2ad822c5045f0ae93beec713c2989e2b5f06323be2b3402d6/merged/host/var/lib/containers/storage/overlay]
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <error>: Failed to unmount: [/var/lib/containers/storage/overlay/33d2c5abb2165ea2ad822c5045f0ae93beec713c2989e2b5f06323be2b3402d6/merged/host/var/lib/containers/storage/overlay]
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <info>: mapped host_mnt=/run/containers/storage to cont_mnt=/host/run/containers/storage
Sep 12 15:11:43 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[13082]: umounthook <debug>: ed630d533d0e30409ba4552b54716e2874c56205fa1692e818b73bdd13fc5060: [/var/lib/containers/storage/overlay/33d2c5abb2165ea2ad822c5045f0ae93beec713c2989e2b5f06323be2b3402d6/merged/host/run/containers/storage] is not a mountpoint. Skipping.
```

**Comment by rhatdan** [comment]
Updated logs now look like
```
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <debug>: prestart container_id:3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3 rootfs:/var/lib/containers/storage/overlay/d271c370ab0ffe23973d7691554e997d7c936928da10ef286be47f5715652710/merged
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <debug>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: Failed to canonicalize path [/var/lib/docker/overlay2]: No such file or directory. Skipping.
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <debug>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: Failed to canonicalize path [/var/lib/docker/overlay]: No such file or directory. Skipping.
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <debug>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: Failed to canonicalize path [/var/lib/docker-latest/overlay2]: No such file or directory. Skipping.
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <debug>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: Failed to canonicalize path [/var/lib/docker-latest/overlay]: No such file or directory. Skipping.
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <debug>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: Failed to canonicalize path [/var/lib/docker-latest/devicemapper]: No such file or directory. Skipping.
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <debug>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: Failed to canonicalize path [/var/lib/docker-latest/containers/]: No such file or directory. Skipping.
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <debug>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: Failed to canonicalize path [/var/lib/containers/storage/lvm]: No such file or directory. Skipping.
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <debug>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: Failed to canonicalize path [/var/lib/containers/storage/devicemapper]: No such file or directory. Skipping.
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <info>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: mapped host_mnt=/var/lib/docker/devicemapper to cont_mnt=/host/var/lib/docker/devicemapper
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <info>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: Unmounted: [/var/lib/containers/storage/overlay/d271c370ab0ffe23973d7691554e997d7c936928da10ef286be47f5715652710/merged/host/var/lib/docker/devicemapper]
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <info>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: mapped host_mnt=/var/lib/docker/containers to cont_mnt=/host/var/lib/docker/containers
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <info>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: mapped host_mnt=/var/lib/containers/storage/overlay to cont_mnt=/host/var/lib/containers/storage/overlay
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <info>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: Unmounted: [/var/lib/containers/storage/overlay/d271c370ab0ffe23973d7691554e997d7c936928da10ef286be47f5715652710/merged/host/var/lib/containers/storage/overlay]
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <info>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: mapped host_mnt=/run/containers/storage to cont_mnt=/host/run/containers/storage
Sep 12 15:24:36 dhcp-10-19-63-214.boston.devel.redhat.com oci-umount[14065]: umounthook <debug>: 3b5979d8c7038edb5081757f873c11493cf716cb92efc0dbe7994b34b9c841e3: [/var/lib/containers/storage/overlay/d271c370ab0ffe23973d7691554e997d7c936928da10ef286be47f5715652710/merged/host/run/containers/storage] is not a mountpoint. Skipping.
```

**Comment by jwhonce** [comment]
@rhatdan Looking much better. Other items to include:

The GUI would not allow me to make these comments in-line:

https://github.com/projectatomic/oci-umount/pull/21/files#diff-c38d5c679d8144acf8977b9a70c3f884R229 does not include the string that failed to strdup() other occurrences do.

Some calls to pr_p* include tailing \n on the fmt parameter others do not.

https://github.com/rhatdan/oci-umount/blob/1f245b4fdad06b3ee4d932ec66f48ea3d7e915ab/src/oci-umount.c#L691 assumes there is only one writer to syslog.  There should be only be one call to pr_error() to report the error.  Again at https://github.com/rhatdan/oci-umount/blob/1f245b4fdad06b3ee4d932ec66f48ea3d7e915ab/src/oci-umount.c#L804


https://github.com/rhatdan/oci-umount/blob/1f245b4fdad06b3ee4d932ec66f48ea3d7e915ab/src/oci-umount.c#L704 should include an identifier to determine which config.json is incorrect.

In general parseBundle() https://github.com/rhatdan/oci-umount/blob/1f245b4fdad06b3ee4d932ec66f48ea3d7e915ab/src/oci-umount.c#L649 pr_p*() calls do not provide context on which bundle is being processed.  Not sure how much that matters for resource allocation but for the configuration parsing would be good to know what is being parsed where.

**Comment by mrunalp** [comment]
Looks fine overall!

File: src/oci-umount.c [hunk 1]
```diff
@@ -365,10 +365,15 @@ static int unmount(char *umount_path, bool submounts_only, const struct mount_in
 
 	if (!submounts_only) {
 		ret = umount2(umount_path, MNT_DETACH);
-		if (!ret)
-			pr_pinfo("Unmounted: [%s]\n", umount_path);
-		else
-			pr_perror("Failed to unmount: [%s]\n", umount_path);
+		if (ret) {
+			if (errno == EINVAL) {
+				pr_pdebug("%s: Not a mount point [%s]\n", id, umount_path);
```
Comment (rhvgoyal) [inline src/oci-umount.c]: How does EINVAL always imply that it is not a valid mount point.  Also before we call unmount() we already check if umount_path() is a mount point or not. So this change does not look right to me.

**Comment by rhvgoyal** [comment]
Can you please break it down in 2-3 patches. First patch just converting some messages to debug messags. Second patch will be for prefixing container ids and third patch will do whatever extra you want to do. Makes it much easier to read and understand what you are planning to do.

File: src/oci-umount.c (hunk 1 above)
Comment (rhatdan) [reply src/oci-umount.c]: Man page.
man umount2 

       EINVAL target is not a mount point.

       EINVAL umount2() was called with MNT_EXPIRE and either MNT_DETACH or MNT_FORCE.

       EINVAL (since Linux 2.6.34)
              umount2() was called with an invalid flag value in flags.

Since the other two would only happen at coding time, most likely the programmer would catch it.

**Comment by rhatdan** [comment]
@jwhonce @rhvgoyal I think I have satisfied all of your requests.

File: src/oci-umount.c [hunk 2]
```diff
@@ -393,8 +393,6 @@ static int unmount(const char *id, char *umount_path, bool submounts_only, const
 		ret = umount2(mnt_table[i].destination, MNT_DETACH);
 		if (!ret)
 			pr_pinfo("%s: Unmounted submount: [%s]\n", id, mnt_table[i].destination);
-		else
-			pr_perror("%s: Failed to unmount submount: [%s]. Skipping.\n", id, mnt_table[i].destination);
```
Comment (rhvgoyal) [inline src/oci-umount.c]: Why this is expected failure?

File: src/oci-umount.c [hunk 3]
```diff
@@ -391,7 +395,13 @@ static int unmount(const char *id, char *umount_path, bool submounts_only, const
 		if (mnt_table[i].parent_mntid != mntid)
 			continue;
 		ret = umount2(mnt_table[i].destination, MNT_DETACH);
-		if (!ret)
+		if (!ret) {
+			if (errno == EINVAL)
+				pr_pdebug("Not a mount point [%s]: %m", umount_path);
```
Comment (rhvgoyal) [inline src/oci-umount.c]: I am still not sure that why this change is required. I think existing code already takes care of it. Before we call umount, we check if this is a mount point or not. And for submounts, we get list from /proc/self/mountinfo, so it has to be a mountpoint. So this does not seem like an issue at all?

**Comment by rhatdan** [comment]
The default config says to umount several mount points that are not mounted, we should not be logging that fact for every container that ever starts.  We expect /var/lib/docker/* to not be mounted when running CRI-O. Even when we run docker, we expect that
/var/lib/docker/overlay or /var/lib/docker/devmapper is not mounted, since it is not being used.  No reason to whine in the log files about it.

**Comment by rhatdan** [comment]
I was seening these messages in the log files when running tests.  That is why I added the code, to shut it up at the info level.

**Comment by rhatdan** [comment]
@rhvgoyal You seem to be commenting on outdated patches.  Could you comment on the latest?

**Comment by rhvgoyal** [comment]
So you want to silently ignore the paths which are not mounted inside container or print a debug message about skipping those paths?

File: src/oci-umount.c (hunk 2 above)
Comment (rhvgoyal) [reply src/oci-umount.c]: This is the commit I am seeing in latest commits. So why are we removing this error message?

**Comment by rhatdan** [comment]
@rhvgoyal yes either way.

**Comment by rhatdan** [comment]
Lets Pick the one that bothers you the most,   and I will revert it so we can get this merged.

File: src/oci-umount.c (hunk 2 above)
Comment (rhatdan) [reply src/oci-umount.c]: This message is coming up in the Hide Outdated, I have no idea why

Comment (rhatdan) [reply src/oci-umount.c]: Ok when I reformated the patches, one removed this code and one put it back,  They should be clean now.

**Comment by rhvgoyal** [comment]
@rhatdan Give me some time. I being at LPC, I am not able to review/test these well. I will review/test this on monday.

**Comment by rhatdan** [comment]
@rhvgoyal PTAL
I have removed the controversial fixes and am now just showing 12 characters in the ID, rather then full access.

**Comment by rhvgoyal** [comment]
So what did you change since last time?

**Comment by rhvgoyal** [comment]
Still crashing. Are you testing all this?

**Comment by rhvgoyal** [comment]
Well new patches are not crashing. I must have tested with old ones. So what has changed?

**Comment by rhvgoyal** [comment]
Let me take that back. Still crashing.

**Comment by rhatdan** [comment]
@rhvgoyal try it now.

**Comment by rhatdan** [comment]
strndupa was crashing, now strndup seems to be working.

**Comment by rhvgoyal** [comment]
LGTM

**Comment by rhvgoyal** [comment]
Which commit needs to be specified to build/merge system? First commit for last commit?

**Comment by rhatdan** [comment]
Last

**Comment by rhatdan** [comment]
Merging

**Comment by rhvgoyal** [comment]
@rhatdan you are using "merge" button of github or using some bot ?

**Comment by rhatdan** [comment]
We don't use a bot for oci-umount.
